packages/angular-cli/plugins/karma.js

```diff
@@ -17,6 +19,25 @@ const init = (config) => {
     progress: config.angularCli.progress
   }
 
+  // add assets
+  if (appConfig.assets) {
+    const assets = typeof appConfig.assets === 'string' ? [appConfig.assets] : appConfig.assets;
+    config.proxies = config.proxies || {};
+    assets.forEach(asset => {
+      const fullAssetPath = path.join(config.basePath, appConfig.root, asset);
+      const isDirectory = fs.lstatSync(fullAssetPath).isDirectory();
+      const filePattern = isDirectory ? fullAssetPath + '/**' : fullAssetPath;
+      const proxyPath = isDirectory ? asset + '/' : asset;
+      config.files.push({
+        pattern: filePattern,
+        included: false,
+        served: true,
+        watched: true
+      });
+      config.proxies['/' + proxyPath] = '/base/' + appConfig.root + '/' + proxyPath;
```

Comment:
I'm no karma expert and worked my way through trial and error really, but according to [the docs](http://karma-runner.github.io/1.0/config/files.html) and my tries, when you add a file to the `files` array, it gets served from `/base/src/path-to-file` only. The proxy adds another endpoint (like a URL rewrite) from `/path-to-file` to `/base/src/path-to-file`.